pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -89,6 +91,14 @@ pub(super) async fn connection_manager_loop_step(
         .timeline
         .subscribe_for_state_updates();
 
+    let mut wait_lsn_status = connection_manager_state
+        .timeline
+        .subscribe_for_wait_lsn_updates();
+
+    // TODO: create a separate config option for discovery request interval
+    let discovery_request_interval = connection_manager_state.conf.lagging_wal_timeout;
```

Comment:
why not `wal_connection_timeout`?

Reply:
Doesn't matter, both options are ok. Just wanted to have some duration without adding a new config parameter (adding new conf can be done in another PR)